src/dxtbx/model/scan.py

```diff
@@ -112,17 +112,67 @@ def from_dict(d, t=None):
         Returns:
             The scan model
         """
+
+        def convert_oscillation_to_vec2(properties_dict):
+
+            """
+            If oscillation is in properties_dict,
+            shared<double> is converted to vec2<double> and
+            oscillation_width is removed (if present) to ensure
+            it is replaced correctly if updating t from d
+            """
+
+            if "oscillation" not in properties_dict:
+                assert "oscillation_width" not in properties_dict
+                return properties_dict
+            if "oscillation_width" in properties_dict:
```

Comment:
`assert "oscillation" in properties_dict`? For consistency with above (since you imply one is possible without the other.